provider/aws.go

```diff
@@ -254,6 +257,39 @@ func (p *AWSProvider) submitChanges(changes []*route53.Change) error {
 	return nil
 }
 
+func limitChangeSet(cs []*route53.Change) []*route53.Change {
+	limit := 4000
+	size := len(cs)
+	if size <= limit {
+		return cs
+	}
+
+	log.Warningf("Initial change batch count is %d which exceeds %d", size, limit)
+
+	countsByName := make(map[string]int)
+	for _, v := range cs {
+		countsByName[*v.ResourceRecordSet.Name]++
+	}
+
+	var newCs = []*route53.Change{}
+
+	remaining := limit
+	for name, count := range countsByName {
+		if (remaining - count) > 0 {
+			remaining -= count
+			for _, v := range cs {
```

Comment:
as a minor optimisation countsByName, could be changesByName, then you don't have to iterate over whole set to find and append the records by name

Reply:
you lost me here - are you saying there already is a changesByName or are you saying for me to move my iteration out into its own function changesByName and call it instead?

Reply:
the idea was to create a map changesByName instead of countsByName: 

```
changesByName := map[string][]*route53.Change
```